operator-framework-core/src/test/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/SSABasedGenericKubernetesResourceMatcherTest.java

```diff
@@ -119,52 +122,132 @@ void addedLabelInDesiredMakesMatchFail() {
     assertThat(matcher.matches(actualConfigMap, desiredConfigMap, mockedContext)).isFalse();
   }
 
-  private <R> R loadResource(String fileName, Class<R> clazz) {
-    return ReconcilerUtils.loadYaml(clazz, SSABasedGenericKubernetesResourceMatcherTest.class,
-        fileName);
-  }
-
   @Test
   @SuppressWarnings("unchecked")
   void sortListItemsTest() {
-    Map<String, Object> nestedMap1 = new HashMap<>();
+    var nestedMap1 = new HashMap<String, Object>();
     nestedMap1.put("z", 26);
     nestedMap1.put("y", 25);
 
-    Map<String, Object> nestedMap2 = new HashMap<>();
+    var nestedMap2 = new HashMap<String, Object>();
     nestedMap2.put("b", 26);
     nestedMap2.put("c", 25);
     nestedMap2.put("a", 24);
 
-    List<Object> unsortedListItems = Arrays.asList(1, nestedMap1, nestedMap2);
-    List<Object> sortedListItems = matcher.sortListItems(unsortedListItems);
-
+    var unsortedListItems = List.<Object>of(1, nestedMap1, nestedMap2);
+    var sortedListItems = matcher.sortListItems(unsortedListItems);
     assertThat(sortedListItems).element(0).isEqualTo(1);
 
-    Map<String, Object> sortedNestedMap1 = (Map<String, Object>) sortedListItems.get(1);
+    var sortedNestedMap1 = (Map<String, Object>) sortedListItems.get(1);
     assertThat(sortedNestedMap1.keySet()).containsExactly("y", "z");
 
-    Map<String, Object> sortedNestedMap2 = (Map<String, Object>) sortedListItems.get(2);
+    var sortedNestedMap2 = (Map<String, Object>) sortedListItems.get(2);
     assertThat(sortedNestedMap2.keySet()).containsExactly("a", "b", "c");
   }
 
   @Test
   @SuppressWarnings("unchecked")
   void testSortMapWithNestedMap() {
-    Map<String, Object> nestedMap = new HashMap<>();
+    var nestedMap = new HashMap<String, Object>();
     nestedMap.put("z", 26);
     nestedMap.put("y", 25);
 
-    Map<String, Object> unsortedMap = new HashMap<>();
+    var unsortedMap = new HashMap<String, Object>();
     unsortedMap.put("b", nestedMap);
     unsortedMap.put("a", 1);
     unsortedMap.put("c", 2);
 
-    Map<String, Object> sortedMap = matcher.sortMap(unsortedMap);
-
+    var sortedMap = matcher.sortMap(unsortedMap);
     assertThat(sortedMap.keySet()).containsExactly("a", "b", "c");
 
-    Map<String, Object> sortedNestedMap = (Map<String, Object>) sortedMap.get("b");
+    var sortedNestedMap = (Map<String, Object>) sortedMap.get("b");
     assertThat(sortedNestedMap.keySet()).containsExactly("y", "z");
   }
+
+  @ParameterizedTest
+  @ValueSource(strings = {"sample-sts-volumeclaimtemplates-desired.yaml",
+      "sample-sts-volumeclaimtemplates-desired-with-status.yaml",
```

Comment:
I'm a little bit more fan of building such resources in java, would be more expressive a bit, but that is jut a nit :)

Reply:
We're using the fabric8 builder classes in our operator tests a lot, but sometimes I think the nested builders are harder to read (heavily depending on the code formatter settings). For example, using `.withNewTemplate()` the nested `.withNewSpec()/endSpec()` are often intended on the same level as the outer `.withNewSpec()/endSpec()` from the STS, which I personally find much harder to read than a well formatted XML file. This can be mitigated by using `.withTemplate(new PodTemplateSpecBuilder()...build())` though.

I just had a look how e.g. `sample-sts-volumeclaimtemplates.yaml` would look like in Java code:
```java
    var actualStatefulSet = new StatefulSetBuilder()
        .withNewMetadata()
        .withName("test")
        .withUid("50913e35-e855-469f-bec6-3e8cd2607ab4")
        .withManagedFields(new ManagedFieldsEntryBuilder()
            .withManager("controller")
            .withOperation("Apply")
            .withApiVersion("apps/v1")
            .withTime("2024-10-24T19:15:25Z")
            .withNewFieldsV1()
             ???
            .endFieldsV1()
            .build())
        .endMetadata()
        .withNewSpec()
        .withReplicas(1)
        .withNewSelector().addToMatchLabels("app", "test-app").endSelector()
        .withServiceName("nginx-service")
        .withTemplate(new PodTemplateSpecBuilder()
            .withNewMetadata().addToLabels("app", "test-app").endMetadata()
            .withNewSpec()
            .addNewContainer()
            .withName("nginx")
            .withImage("nginx:1.17.0")
            .addNewPort().withContainerPort(80).endPort()
            .addNewVolumeMount()
            .withName("persistent-storage")
            .withMountPath("/usr/share/nginx/html")
            .endVolumeMount()
            .endContainer()
            .endSpec()
            .build())
        .addToVolumeClaimTemplates(new PersistentVolumeClaimBuilder()
            .withNewMetadata().withName("persistent-storage").endMetadata()
            .withNewSpec()
            .addToAccessModes("ReadWriteOnce")
            .withNewResources()
            .addToRequests("storage", new Quantity("1Gi"))
            .endResources()
            .withStorageClassName("standard")
            .withVolumeMode("Filesystem")
            .endSpec()
            .withNewStatus()
            .withPhase("Pending")
            .endStatus()
            .build())
        .endSpec()
        .build();
```
I gave up on the managed fields. Without looking up an example in the fabric8 client I have no idea how to use that builder. I already had to guess how to write that correctly in XML, by adapting this from another resource file :sweat_smile:

So even as a heavy user of those builder classes, I think for these complex resources XML files are easier to write and even to compare (IDEA -> mark two files -> diff).